test/preview/document_stores/test_writer_integration.py

```diff
@@ -0,0 +1,24 @@
+import collections
+
+import pytest
+
+from haystack.preview import Document, Pipeline
+from haystack.preview.components.writers.document_writer import DocumentWriter
+from haystack.preview.document_stores import MemoryDocumentStore
+
+
+@pytest.mark.integration
+def test_writer_adds_documents_to_store():
```

Comment:
I believe this test is not necessary as we're not even testing the `DocumentWriter` really.
If we want to test that the component does what we're supposed to do we just need to verify that `self.store.write_documents` is called when executing `run`. 
The actual writing is not this component's responsibility.
Something like this should work.

```
@pytest.mark.unit
def test_document_writer_run():
    mock_store = store(MagicMock)()
    writer = DocumentWriter()
    writer.store = mock_store

    data = writer.input(documents=[Document(content="This is the text of a document.")])
    writer.run(data=data)

    assert mock_store.write_documents.called```